conda/core/package_cache_data.py

```diff
@@ -71,10 +72,10 @@
 class PackageCacheType(type):
     """This metaclass does basic caching of PackageCache instance objects."""
 
-    def __call__(cls, pkgs_dir):
+    def __call__(cls, pkgs_dir: str | os.PathLike | Path):
         if isinstance(pkgs_dir, PackageCacheData):
             return pkgs_dir
-        elif pkgs_dir in PackageCacheData._cache_:
+        elif (pkgs_dir := str(pkgs_dir)) in PackageCacheData._cache_:
```

Comment:
I'm still a bit green with `:=` does this assignment propagate to the block under `else` or is it local to this block only?

Reply:
yes, it propagates

it's a normal local variable, so any subsequent code can access it

the walrus operator did not change how scopes work